File: Python/9.Palindrome Number.py
#判断一个数是否为回文数
import math
class Solution(object):
    def isPalindrome(self, x):
        if x<0:
            return False
        p,ten = 1,10
        while x>=ten:
            p+=1
            ten *= 10
        i = 0
        while i<=p//2:
            m = x//(math.pow(10,i) if i!=0 else 1)%10
            n = int(x//(math.pow(10,p-i-1))%10)
            if m==n:
                i+=1
            else:
                return False
        return True

s = Solution()
print(s.isPalindrome(-2147447412))

File: C++/Math/9.Palindrome Number.cpp
/*
判断一个整型数是不是回文的。
*/
/*
思路：取得后一半和前一半比较。
 */
#include <iostream>
#include <limits.h>
using namespace std;

class Solution {
public:
	bool isPalindrome(int x) {
		//负数和0结尾的数一定不是回文的。
        if(x<0|| (x!=0 &&x%10==0)) return false;
        int sum=0;
        while(x>sum)
        {
            sum = sum*10+x%10;
            x = x/10;
        }
        return (x==sum)||(x==sum/10);
    }
};
int main()
{
	Solution s;
 	cout<<s.isPalindrome(123);
}

